.gitlab-ci.yml

```diff
@@ -12,5 +17,107 @@ include:
+test:
+  stage: test
+  image: golang:1.17
+  before_script:
+    - apt-get update && apt-get install -yyq libssl-dev libglib2.0-dev dbus clang-format-9
+    - GO111MODULE=off go get -u github.com/jstemmer/go-junit-report
+  script:
+    - git ls-tree -r --name-only HEAD | grep -v vendor/ | grep '\.[ch]$' | xargs clang-format-9 -i
+    - make coverage
+    - make
+    - mkdir -p tests/unit-coverage && find . -name 'coverage.txt' -exec cp --parents {} ./tests/unit-coverage \;
+    - tar -cvf $CI_PROJECT_DIR/unit-coverage.tar tests/unit-coverage
+  tags:
+    - mender-qa-slave
+  artifacts:
+    expire_in: 2w
+    reports:
+      junit: report.xml
+    paths:
+      - unit-coverage.tar
+
+publish:tests:
+  stage: publish
+  image: golang:1.17-alpine3.14
+  dependencies:
+    - test
+  before_script:
+    # Install dependencies
+    - apk add --no-cache git
+    - GO111MODULE=off go get -u github.com/mattn/goveralls
+    # Coveralls env variables:
+    #  According to https://docs.coveralls.io/supported-ci-services
+    #  we should set CI_NAME, CI_BUILD_NUMBER, etc. But according
+    #  to goveralls source code (https://github.com/mattn/goveralls)
+    #  many of these are not supported. Set CI_BRANCH,
+    #  and pass few others as command line arguments.
+    #  See also https://docs.coveralls.io/api-reference
+    - export CI_BRANCH=${CI_COMMIT_BRANCH}
+  script:
+    - tar -xvf unit-coverage.tar
+    - goveralls
+      -repotoken ${COVERALLS_TOKEN}
+      -service gitlab-ci
+      -jobid $(git rev-parse HEAD)
+      -covermode set
+      -flagname unittests
+      -parallel
+      -coverprofile $(find tests/unit-coverage -name 'coverage.txt' | tr '\n' ',' | sed 's/,$//')
+
+generate-qa-trigger:
+  image: python:alpine
+  stage: trigger_prep
+  rules:
+    - if: '$CI_COMMIT_BRANCH =~ /^(master|[0-9]+\.[0-9]+\.x)$/'
+    - if: '$CI_PIPELINE_SOURCE == "pipeline"'
+      when: never
+    - if: '$CI_PIPELINE_SOURCE == "schedule"'
+      when: never
+  before_script:
+    - apk add --no-cache git
+    - pip3 install pyyaml
+    - wget -q https://raw.githubusercontent.com/mendersoftware/mender-qa/master/scripts/generate_client_publish_job.py
+  script:
+    - python generate_client_publish_job.py --trigger ${CI_PROJECT_NAME} --version ${CI_COMMIT_REF_NAME} --filename gitlab-ci-client-qemu-publish-job.yml
+  artifacts:
+    paths:
+      - gitlab-ci-client-qemu-publish-job.yml
+
+trigger:mender-qa:
+  stage: trigger
+  rules:
+    - if: '$CI_COMMIT_BRANCH =~ /^(master|[0-9]+\.[0-9]+\.x)$/'
+    # the following is to prevent an endless loop of qa pipelines caused by downstream pipelines
+    - if: '$CI_PIPELINE_SOURCE == "pipeline"'
+      when: never
+    - if: '$CI_PIPELINE_SOURCE == "schedule"'
+      when: never
+  trigger:
+    include:
+      - artifact: gitlab-ci-client-qemu-publish-job.yml
+        job: generate-qa-trigger
+
 # Keep golang version aligned with latest yocto release
 image: golang:1.17-alpine3.16
+
+trigger:mender-dist-packages:
```

Comment:
This we don't need. As per today we will trigger the builds from the main mender-qa pipeline